### PR TITLE
Added noll/multisteno

### PR DESCRIPTION
**Pull request by igniuss**
## Description
Added a new keyboard, so I can use the QMK configurator,

the main problem is that I do not _own_ this keyboard, but it hasn't been updated since QMK 1.1.1, so I'm not sure what the etiquette for that is.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Added [noll/multisteno](https://nolltronics.com/product/multisteno/) as a keyboard

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/noll/multisteno/chconf.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/noll/ecosteno/chconf.h -r platforms/chibios/common/configs/chconf.h`
+ */
```
Comment (waffle87) [inline keyboards/noll/multisteno/chconf.h:20]:
```suggestion
// Copyright 2024 QMK
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/noll/multisteno/config.h
Comment (waffle87) [inline keyboards/noll/multisteno/config.h:1]: This file's contents need to migrated to `info.json`, and this file removed.

File: keyboards/noll/multisteno/halconf.h
Comment (waffle87) [inline keyboards/noll/multisteno/halconf.h:1]: This file should be removed.

File: keyboards/noll/multisteno/info.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    },
+    "processor": "STM32F103",
+    "stenography": {
+        "enabled": true,
+        "protocol": "geminipr"
+    },
+    "tapping": {
+        "term": 137
+    },
+    "url": "",
```
Comment (waffle87) [inline keyboards/noll/multisteno/info.json:25]:
```suggestion
    "features": {
        "bootmagic": true,
        "mousekey": true,
        "extrakey": true,
        "stenography": true
    },
    "diode_direction": "COL2ROW",
    "processor": "STM32F103",
    "stenography": {
        "protocol": "geminipr"
    },
```

**Comment by keyboard-magpie** [comment]
Configurator is a very poor tool for such a keyboard. 

Technically you can PR someone else's code but copyright needs to be preserved and added to by you.

However, the code you have submitted requires significant updates to be valid. Some insight to this can be seen in why the Lint checks and CI checks are failing below.

You also need to go through the PR checklist carefully, and ensure you have actioned all the relevant points in it, and then check the box in the Checklist above. Only then will it be considered ready for initial review.

File: keyboards/noll/multisteno/keyboard.c
Comment (waffle87) [inline keyboards/noll/multisteno/keyboard.c:1]: This file should be renamed to `multisteno.c`.

File: keyboards/noll/multisteno/keyboard.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * You should have received a copy of the GNU General Public License 
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+ */
+#include <ch.h>
+#include <hal.h>
+#include "util.h"
+#include "quantum.h"
+
+#ifdef BOARD_STM32_F103_STM32DUINO
+
```
Comment (waffle87) [inline keyboards/noll/multisteno/keyboard.c:22]: There's probably a better way to do all this. What exactly are you trying to achieve here?

File: keyboards/noll/multisteno/keymaps/default/keymap.c
Comment (waffle87) [inline keyboards/noll/multisteno/keymaps/default/keymap.c:1]: Default keymaps are meant to be pristine, ie. only include what's needed for basic functionality of the keyboard, meaning combos and other custom code should be removed.

File: keyboards/noll/multisteno/keymaps/txbolt/keymap.c
Comment (waffle87) [inline keyboards/noll/multisteno/keymaps/txbolt/keymap.c:1]: As above.

File: keyboards/noll/multisteno/keymaps/txbolt/rules.mk [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE = no	# Mouse keys
+EXTRAKEY_ENABLE = no	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+COMBO_ENABLE = yes		# Enable Key Combos
+STENO_ENABLE = yes		# Stenography keys
+STENO_PROTOCOL = txbolt	# geminipr or txbolt
+MOUSEKEY_ENABLE = no	# Mouse keys
```
Comment (waffle87) [inline keyboards/noll/multisteno/keymaps/txbolt/rules.mk:20]:
```suggestion
STENO_PROTOCOL = txbolt
```

File: keyboards/noll/multisteno/mcuconf.h [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/noll/ecosteno/chconf.h -r platforms/chibios/STM32_F103_STM32DUINO/configs/mcuconf.h`
+ */
```
Comment (waffle87) [inline keyboards/noll/multisteno/mcuconf.h:20]:
```suggestion
// Copyright 2024 QMK
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/noll/multisteno/multisteno.c
Comment (waffle87) [inline keyboards/noll/multisteno/multisteno.c:1]: This file should be removed.

File: keyboards/noll/multisteno/multisteno.h
Comment (waffle87) [inline keyboards/noll/multisteno/multisteno.h:1]: This file should be removed.

File: keyboards/noll/multisteno/readme.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+Multisteno
+===
```
Comment (waffle87) [inline keyboards/noll/multisteno/readme.md:2]:
```suggestion
# Multisteno
```

File: keyboards/noll/multisteno/readme.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+![Multisteno](https://nolltronics.com/wp-content/uploads/2022/05/DSC03181-Multi-Main.jpg)
```
Comment (waffle87) [inline keyboards/noll/multisteno/readme.md:4]: This image should be rehosted on Imgur.

File: keyboards/noll/multisteno/readme.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+* Hardware Supported: Multisteno v1.1.2X
+* Hardware Availability: [Nolltronics](https://nolltronics.com/product/multisteno)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make noll/multisteno:default
+
+Flashing example for this keyboard:
+
+    dfu-util.exe -d 1eaf:0003 -a 2 -D "noll_multisteno_default.bin"
```
Comment (waffle87) [inline keyboards/noll/multisteno/readme.md:16]:
```suggestion
Make example for this keyboard (after setting up your build environment):

    make noll/multisteno:default

Flashing example for this keyboard:

    make noll/multisteno:default:flash
```

File: keyboards/noll/multisteno/rules.mk [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+# project specific files
+SRC =	keyboard.c
+
+EXTRAKEY_ENABLE = no	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+COMBO_ENABLE = yes		# Enable Key Combos
```
Comment (waffle87) [inline keyboards/noll/multisteno/rules.mk:13]:
```suggestion
# This file intentionally left blank
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/noll/multisteno/keymaps/default/keymap.c
Comment (igniuss) [reply keyboards/noll/multisteno/keymaps/default/keymap.c:1]: Yeah sorry about that, it's originally based off of one of their older keyboards, which has much fewer keys; so it used a bunch of combos. We don't need that stuff on here though!

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/noll/multisteno/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+        case (SYMBOL):
+          // both LEDs on for Symbol
+          palSetPad(GPIOA, 1);
+          palSetPad(GPIOA, 0);
+          break;
+        default:
+          palClearPad(GPIOA, 1);
+          palClearPad(GPIOA, 0);
+          break;
+    }
```
Comment (zvecr) [inline keyboards/noll/multisteno/keymaps/default/keymap.c:98]:
```suggestion
    switch (get_highest_layer(state)) {
        case (STENO):
          //green LED on
          // red LED off
          gpio_write_pin_high(A1);
          gpio_write_pin_low(A0);
          break;
        case (QWERTY):
          // red LED on
          // green LED off
          gpio_write_pin_high(A0);
          gpio_write_pin_low(A1);
          break;
        case (QWERTY_CAPS):
          // same as QWERTY
          gpio_write_pin_high(A0);
          gpio_write_pin_low(A1);
          break;
        case (NKRO): 
          gpio_write_pin_low(A0);
          gpio_write_pin_low(A1);
          break;
        case (SYMBOL):
          // both LEDs on for Symbol
          gpio_write_pin_high(A0);
          gpio_write_pin_high(A1);
          break;
        default:
          gpio_write_pin_low(A0);
          gpio_write_pin_low(A1);
          break;
    }
```

File: keyboards/noll/multisteno/keymaps/txbolt/keymap.c [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+        case (SYMBOL):
+          // both LEDs on for Symbol
+          palSetPad(GPIOA, 1);
+          palSetPad(GPIOA, 0);
+          break;
+        default:
+          palClearPad(GPIOA, 1);
+          palClearPad(GPIOA, 0);
+          break;
+    }
```
Comment (zvecr) [inline keyboards/noll/multisteno/keymaps/txbolt/keymap.c:98]:
```suggestion
    switch (get_highest_layer(state)) {
        case (STENO):
          //green LED on
          // red LED off
          gpio_write_pin_high(A1);
          gpio_write_pin_low(A0);
          break;
        case (QWERTY):
          // red LED on
          // green LED off
          gpio_write_pin_high(A0);
          gpio_write_pin_low(A1);
          break;
        case (QWERTY_CAPS):
          // same as QWERTY
          gpio_write_pin_high(A0);
          gpio_write_pin_low(A1);
          break;
        case (NKRO): 
          gpio_write_pin_low(A0);
          gpio_write_pin_low(A1);
          break;
        case (SYMBOL):
          // both LEDs on for Symbol
          gpio_write_pin_high(A0);
          gpio_write_pin_high(A1);
          break;
        default:
          gpio_write_pin_low(A0);
          gpio_write_pin_low(A1);
          break;
    }
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/noll/multisteno/multisteno.c [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+#define LED_OFF()
+#define LED_TGL()
+#endif
+
+void keyboard_pre_init_user(void){
+    // Initialize LED pins to correct setting
+    palSetPadMode(GPIOA, 0, PAL_MODE_OUTPUT_PUSHPULL); // 6U correct
+    palSetPadMode(GPIOA, 1, PAL_MODE_OUTPUT_PUSHPULL); // 6U correct
+
+}
```
[the remark this answers is not in the document]
Comment (igniuss) [reply keyboards/noll/multisteno/multisteno.c:43]: This seems to produce some errors on builds when I change it locally, is there an include I need or something for these functions to show up? 👀

**Comment by igniuss** [comment]
I just want to thank you so much @zvecr for the time and patience; doing all of this is very overwhelming, and your assistance is super appreciated 😍

File: keyboards/noll/multisteno/info.json [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+    "bootloader": "stm32duino",
+    "features": {
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "combo": {
+        "term": 35
+    },
```
Comment (waffle87) [inline keyboards/noll/multisteno/info.json:14]:
```suggestion
```

File: keyboards/noll/multisteno/info.json [hunk 14]
```diff
@@ -0,0 +1,80 @@
+{
+    "manufacturer": "Noll Electronics LLC",
+    "keyboard_name": "Nolltronics Multisteno",
+    "maintainer": "Noll Electronics LLC",
+    "bootloader": "stm32duino",
+    "features": {
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (waffle87) [inline keyboards/noll/multisteno/info.json:10]:
```suggestion
        "nkro": true,
        "steno": true
```

File: keyboards/noll/multisteno/info.json [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+    },
+    "combo": {
+        "term": 35
+    },
+    "processor": "STM32F103",
+    "stenography": {
+        "enabled": true,
```
Comment (waffle87) [inline keyboards/noll/multisteno/info.json:17]:
```suggestion
```

File: keyboards/noll/multisteno/info.json [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+    "manufacturer": "Noll Electronics LLC",
+    "keyboard_name": "Nolltronics Multisteno",
+    "maintainer": "Noll Electronics LLC",
+    "bootloader": "stm32duino",
+    "features": {
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "stenography": true
```
Comment (waffle87) [inline keyboards/noll/multisteno/info.json:11]: This does not work. Please apply the suggested changes, or at least verify that the changes you make on your own produce the expected functionality.
```suggestion
        "steno": true
```

File: keyboards/noll/multisteno/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+        gpio_write_pin_high(A0);
+        gpio_write_pin_low(A1);
+        break;
+      case (QWERTY_CAPS):
+        // same as QWERTY
+        gpio_write_pin_high(A0);
+        gpio_write_pin_low(A1);
+        break;
+      case (NKRO): 
+            gpio_write_pin_low(A0);
```
Comment (waffle87) [inline keyboards/noll/multisteno/keymaps/default/keymap.c:86]:
```suggestion
        gpio_write_pin_low(A0);
```

File: keyboards/noll/multisteno/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+        break;
+      case (SYMBOL):
+        // both LEDs on for Symbol
+        gpio_write_pin_high(A0);
+        gpio_write_pin_high(A1);
+        break;
+      default:
+        gpio_write_pin_low(A0);
+        gpio_write_pin_low(A1);
+        break;
```
Comment (waffle87) [inline keyboards/noll/multisteno/keymaps/default/keymap.c:97]:
```suggestion
```

File: keyboards/noll/multisteno/keymaps/txbolt/keymap.c [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+          break;
+        case (SYMBOL):
+          // both LEDs on for Symbol
+          gpio_write_pin_high(A0);
+          gpio_write_pin_high(A1);
+          break;
+        default:
+          gpio_write_pin_low(A0);
+          gpio_write_pin_low(A1);
+          break;
```
Comment (waffle87) [inline keyboards/noll/multisteno/keymaps/txbolt/keymap.c:97]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/noll/multisteno/info.json (hunk 16 above)
Comment (igniuss) [reply keyboards/noll/multisteno/info.json:11]: So sorry, I just realized I was running my linter in the wrong QMK directory 🤦🏻

**Review by waffle87 — APPROVED** [review]

File: keyboards/noll/multisteno/rules.mk
Comment (fauxpark) [inline keyboards/noll/multisteno/rules.mk:1]: This file can be deleted.

File: keyboards/noll/multisteno/info.json
Comment (fauxpark) [inline keyboards/noll/multisteno/info.json:1]: This file should be renamed to `keyboard.json`.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**